GCC 13 build problems (this is for tracking when the move from GCC 10/11 happens)

I have been building with an upcoming release of GCC and this may affect some people. Just getting it out there as eventually we will be at GCC 13 or later.

1.  **src/libm/math/kf_rem_pio2.c**

```
src/libm/math/kf_rem_pio2.c: In function '__kernel_rem_pio2f':
src/libm/math/kf_rem_pio2.c:186:24: error: 'fq' may be used uninitialized [-Werror=maybe-uninitialized]
  186 |                 fw = fq[0]-fw;
      |                      ~~^~~
src/libm/math/kf_rem_pio2.c:63:26: note: 'fq' declared here
   63 |         float z,fw,f[20],fq[20],q[20];
      |                          ^~
cc1: all warnings being treated as errors
```
Edit `libs/Makefile FLAGS  -Wno-error=maybe-uninitialized`

2.  **memcheck.c**
```
In file included from ../../../code/software/libs/build/include/stdio.h:20,
                 from memcheck.c:16:
In function 'show_banner',
    inlined from 'main' at memcheck.c:400:3:
../../../code/software/libs/build/include/printf.h:59:16: error: array subscript 0 is outside array bounds of 'uint32_t[0]' {aka 'long unsigned int[]'} [-Werror=array-bounds=]
   59 | #define printf printf_
memcheck.c:346:5: note: in expansion of macro 'printf'
  346 |     printf("* Firmware reports %8ld bytes total contiguous memory *\n", *FW_MEMSIZE);
      |     ^~~~~~
In function 'main':
cc1: note: source object is likely at address zero
cc1: all warnings being treated as errors
```
Edit `memcheck/Makefile CFLAGS -Wno-array-bounds`

3. **updateflash.c**
```
updateflash.c: In function 'main':
updateflash.c:214:25: error: array subscript 256 is outside array bounds of 'void[4]' [-Werror=array-bounds=]
  214 |         if (version_info->major != 0) {
      |                         ^~
updateflash.c:29:14: note: at offset 1024 into object '_end' of size 4
   29 | extern void *_end;
      |              ^~~~
updateflash.c:218:33: error: array subscript 256 is outside array bounds of 'void[4]' [-Werror=array-bounds=]
  218 |                     version_info->is_extdata ? "Required" : "Not required");
      |                                 ^~
updateflash.c:29:14: note: at offset 1024 into object '_end' of size 4
   29 | extern void *_end;
      |              ^~~~
updateflash.c:216:54: error: array subscript 256 is outside array bounds of 'void[4]' [-Werror=array-bounds=]
  216 |                     version_info->major, version_info->minor,
      |                                                      ^~
updateflash.c:29:14: note: at offset 1024 into object '_end' of size 4
   29 | extern void *_end;
      |              ^~~~
updateflash.c:220:29: error: array subscript 256 is outside array bounds of 'void[4]' [-Werror=array-bounds=]
  220 |             if (version_info->is_huge) {
      |                             ^~
updateflash.c:29:14: note: at offset 1024 into object '_end' of size 4
   29 | extern void *_end;
      |              ^~~~
cc1: all warnings being treated as errors
```
Edit  `updateflash/Makefile CFLAGS -Wno-array-bounds`

--- a/code/firmware/rosco_m68k_firmware/Makefile
+++ b/code/firmware/rosco_m68k_firmware/Makefile
@@ -23,10 +23,15 @@ SIZE=m68k-elf-size
 NM=m68k-elf-nm
 RM=rm -f
 
-# If this is GCC 12, add flag --param=min-pagesize=0
+# GCC-version-specific settings
 ifneq ($(findstring GCC,$(shell $(CC) --version 2>/dev/null)),)
 CC_VERSION:=$(shell $(CC) -dumpfullversion)
-ifeq ($(firstword $(subst ., ,$(CC_VERSION))),12)
+CC_MAJOR:=$(firstword $(subst ., ,$(CC_VERSION)))
+# If this is GCC 12 or 13, add flag --param=min-pagesize=0 to CFLAGS
+ifeq ($(CC_MAJOR),12)
+CFLAGS+=--param=min-pagesize=0
+endif
+ifeq ($(CC_MAJOR),13)
 CFLAGS+=--param=min-pagesize=0
 endif
 endif

--- a/code/firmware/rosco_m68k_firmware/stage2/Makefile
+++ b/code/firmware/rosco_m68k_firmware/stage2/Makefile
@@ -21,11 +21,16 @@ LD=m68k-elf-ld
 AS=vasmm68k_mot
 RM=rm -f
 
-# If this is GCC 12, add flag --param=min-pagesize=0
+# GCC-version-specific settings
 ifneq ($(findstring GCC,$(shell $(CC) --version 2>/dev/null)),)
 CC_VERSION:=$(shell $(CC) -dumpfullversion)
-ifeq ($(firstword $(subst ., ,$(CC_VERSION))),12)
-BASE_CFLAGS+=--param=min-pagesize=0
+CC_MAJOR:=$(firstword $(subst ., ,$(CC_VERSION)))
+# If this is GCC 12 or 13, add flag --param=min-pagesize=0 to CFLAGS
+ifeq ($(CC_MAJOR),12)
+CFLAGS+=--param=min-pagesize=0
+endif
+ifeq ($(CC_MAJOR),13)
+CFLAGS+=--param=min-pagesize=0
 endif
 endif
 

--- a/code/software/libs/src/libm/include.mk
+++ b/code/software/libs/src/libm/include.mk
@@ -12,6 +12,16 @@ OBJECTS := $(OBJECTS) $(LIBOBJECTS)
 INCLUDES := $(INCLUDES) $(DIR)/include/*
 LIBS := $(LIBS) $(DIR)/$(BINARY)
 
+# GCC-version-specific settings
+ifneq ($(findstring GCC,$(shell $(CC) --version 2>/dev/null)),)
+CC_VERSION:=$(shell $(CC) -dumpfullversion)
+CC_MAJOR:=$(firstword $(subst ., ,$(CC_VERSION)))
+# If this is GCC 13, add flag -Wno-error=maybe-uninitialized
+ifeq ($(CC_MAJOR),13)
+CFLAGS+=-Wno-error=maybe-uninitialized
+endif
+endif
+
 $(DIR)/$(BINARY): $(LIBOBJECTS)
 	$(AR)	$(ARFLAGS) rs $@ $^
 	$(RANLIB) $@

